src/pymor/models/interface.py

```diff
@@ -109,6 +109,74 @@ def _compute_output(self, solution, mu=None, **kwargs):
         else:
             return self.output_functional.apply(solution, mu=mu).to_numpy()
 
+    def _compute_solution_d_mu_single_direction(self, parameter, index, solution, mu=None, **kwargs):
+        """Compute the partial derivative of the solution w.r.t. a parameter index
+
+        Parameters
+        ----------
+        parameter
+            parameter for which to compute the sensitivity
+        index
+            parameter index for which to compute the sensitivity
+        solution
+            Internal model state for the given |Parameter value|.
+        mu
+            |Parameter value| for which to solve
+
+        Returns
+        -------
+        The sensitivity of the solution as a |VectorArray|.
+        """
+        raise NotImplementedError
+
+    def _compute_solution_d_mu(self, solution, mu=None, **kwargs):
+        """Compute all partial derivative of the solution w.r.t. a parameter index
+
+        Parameters
+        ----------
+        solution
+            Internal model state for the given |Parameter value|.
+        mu
+            |Parameter value| for which to solve
+
+        Returns
+        -------
+        A dict of all partial sensitivities of the solution.
+        """
+        sensitivities = {}
+        for (parameter, size) in sorted(self.parameters.items()):
```

Comment:
```suggestion
        for (parameter, size) in self.parameters.items():
```

`Parameters` are sorted now.